src/utils.js

```diff
@@ -1,3 +1,65 @@
+const NON_STRING_PLACEHOLDER = "__REACT_EMOJI_RENDER_PLACEHOLDER__";
+
 export function escapeStringToBeUsedInRegExp(string) {
   return string.replace(/[.*+?^${}()|[\]\\]/g, "\\$&"); // $& means the whole matched string
 }
+
+export function stripNonStringElements(children) {
+  const stripped = children
+    .map(x => (typeof x === "string" ? x : NON_STRING_PLACEHOLDER))
+    .join("");
+  return [stripped, children.filter(x => typeof x !== "string")];
```

Comment:
Let's be a bit more explicit on the variables names.

```suggestion
  const strippedChildren = children
    .map(child => (typeof child === "string" ? child : NON_STRING_PLACEHOLDER))
    .join("");
  const elements = children.filter(x => typeof x !== "string");
  return [strippedChildren, elements];
```